packages/hoppscotch-app/helpers/curl/sub_helpers/url.ts

```diff
@@ -0,0 +1,92 @@
+import parser from "yargs-parser"
+import { pipe } from "fp-ts/function"
+import * as O from "fp-ts/Option"
+import { getDefaultRESTRequest } from "~/newstore/RESTSession"
+import { stringArrayJoin } from "~/helpers/functional/array"
+
+const defaultRESTReq = getDefaultRESTRequest()
+
+/**
+ * Processes URL string and returns the URL object
+ * @param parsedArguments Parsed Arguments object
+ * @returns URL object
+ */
+export function parseURL(parsedArguments: parser.Arguments) {
+  return pipe(
+    parsedArguments._[1],
+    O.fromNullable,
+    O.map((u) => u.toString().replace(/["']/g, "")),
+    O.map((u) => u.trim()),
+    O.chain((u) =>
+      pipe(
+        /^[^:\s]+(?=:\/\/)/.exec(u),
+        O.fromNullable,
+        O.map((p) => p[2]),
```

Comment:
Don't we have to make sure the array has enough length to make sure there are no errors.